yarn.lock

```diff
@@ -0,0 +1,354 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```

Comment:
why this file is required?

Reply:
that is a lock file generated by yarn (package manager) install.
Its similar to a package-lock.json file generated when one uses npm install
It is meant to lock the exact version of the package/modules used during development.

https://robertcooper.me/post/how-yarn-lock-files-work-and-upgrading-dependencies

when these lock files are not present the package.json file typically has ^ sign prefixed to version number of each dependency, which locks only the major number of the version.

eg 
`"react": "^17.0.2"`,
accepts:
`"react": "^17.0.2"`,
`"react": "^17.0.2"`,
`"react": "^17.0.2"`,